internal/pipe/nexus_raw/nexus_raw.go

```diff
@@ -0,0 +1,145 @@
+package nexus_raw
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/apex/log"
+	"github.com/goreleaser/goreleaser/internal/artifact"
+	"github.com/goreleaser/goreleaser/internal/pipe"
+	"github.com/goreleaser/goreleaser/internal/semerrgroup"
+	"github.com/goreleaser/goreleaser/pkg/config"
+	"github.com/goreleaser/goreleaser/pkg/context"
+)
+
+type Pipe struct{}
+
+func (Pipe) String() string {
+	return "Sonatype Nexus Raw Repositories"
+}
+
+func (Pipe) Publish(ctx *context.Context) error {
+	if len(ctx.Config.NexusRaws) == 0 {
+		return pipe.Skip("sonatype nexus section is not configured")
+	}
+	for _, instance := range ctx.Config.NexusRaws {
+		if skipErr := checkConfig(&instance); skipErr != nil {
+			return pipe.Skip(skipErr.Error())
+		}
+	}
+
+	if ctx.SkipPublish {
+		return pipe.ErrSkipPublishEnabled
+	}
+
+	for _, nexus := range ctx.Config.NexusRaws {
+		if err := upload(ctx, &nexus); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func upload(ctx *context.Context, nexus *config.NexusRaw) error {
+	g := semerrgroup.New(ctx.Parallelism)
+	for _, art := range ctx.Artifacts.Filter(
+		artifact.Or(
+			artifact.ByType(artifact.UploadableArchive),
+			artifact.ByType(artifact.UploadableBinary),
+			artifact.ByType(artifact.Checksum),
+			artifact.ByType(artifact.Signature),
+			artifact.ByType(artifact.LinuxPackage),
+		),
+	).List() {
+		art := art
+		g.Go(func() error {
+			return uploadAsset(nexus, art)
+		})
+	}
+	return g.Wait()
+}
+
+func uploadAsset(nexus *config.NexusRaw, artif *artifact.Artifact) error {
+	var b bytes.Buffer
+	log.WithField("file", artif.Path).WithField("name", artif.Name).Info("uploading to nexus")
+	w := multipart.NewWriter(&b)
+	if err := createForm(w, artif.Path, artif.Name, nexus.Directory); err != nil {
+		return err
+	}
+
+	url := fmt.Sprintf("%s/service/rest/v1/components?repository=%s", nexus.Url, nexus.Repository)
+	req, err := http.NewRequest("POST", url, &b)
+	if err != nil {
+		return err
+	}
+	req.Header.Set("Content-Type", w.FormDataContentType())
+	if nexus.Username != "" {
+		req.SetBasicAuth(nexus.Username, nexus.Password)
+	}
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+	if res.StatusCode != http.StatusNoContent {
+		var bodyContents string
+		body, err := ioutil.ReadAll(res.Body)
+		if err == nil {
+			bodyContents = string(body)
+		}
+		return pipe.Skip(fmt.Sprintf("bad status: %s, body: %s", res.Status, bodyContents))
+	}
+	return nil
+}
+
+func createForm(w *multipart.Writer, artifPath, artifName, nexusDir string) error {
```

Comment:
errors inside this function shouldn't be `skip errors`.

skip errs are used only to skip a pipe if some configuration is missing, other that that, regular errors should be used.